ledger/complete/wal/checkpointer.go

```diff
@@ -589,6 +655,8 @@ func readCheckpoint(f *os.File) ([]*trie.MTrie, error) {
 		return readCheckpointV4(f)
 	case VersionV5:
```

Comment:
why do we need to keep v1-v4? since spork forces all binaries to upgrade, these are just deadcode / bugs when to happen? would be great if we can delete these (and their associate functions) first since I have to understand what the system currently does in order to provide useful review feedback.

Reply:
yeah, we can probably delete v1-v4. we can do in separate PR. just to double check with @fxamacker @ramtinms

Reply:
I understand but I think we may need to handle decoding of all previous checkpoint versions.  For example, DPS or some third-party software might need to load old checkpoint file formats.

Reply:
third party software can use older version of flow-go to read old checkpoints?  also, the code is still in git history ...

Reply:
might need further discussion, [tracked in this issue](https://github.com/dapperlabs/flow-go/issues/6404)